### PR TITLE
[go_router] Added missing implementation for the routerNeglect parameter in GoRouter

### PR DESCRIPTION
**Pull request by tolo**
The `GoRouter` constructor parameter `routerNeglect` currently has no effect. This PR provides an implementation for this parameter, in line with the original implementation, i.e. _"This will suppress history for all navigation made using go_router"_.

*List which issues are fixed by this PR. You must list at least one issue.*
* flutter/flutter#119398 

## Pre-launch Checklist

- [x] I read the [[Contributor Guide](https://github.com/flutter/packages/blob/main/CONTRIBUTING.md)] and followed the process outlined there for submitting PRs.
- [x] I read the [[Tree Hygiene](https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md)] page, which explains my responsibilities.
- [x] I read and followed the [[relevant style guides](https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style)] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [[CLA](https://cla.developers.google.com/)].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I [[linked to at least one issue that this PR fixes](https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#overview)] in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [[pub versioning philosophy](https://dart.dev/tools/pub/versioning)], or this PR is [[exempt from version changes](https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#version)].
- [x] I updated `CHANGELOG.md` to add a description of the change, [[following repository CHANGELOG style](https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog-style)], or this PR is [[exempt from CHANGELOG changes](https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog)].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [[test-exempt](https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#tests)].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [[Discord](https://github.com/flutter/flutter/blob/master/docs/contributing/Chat.md)].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/blob/master/docs/contributing/Chat.md
[linked to at least one issue that this PR fixes]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#overview
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#version
[following repository CHANGELOG style]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog-style
[exempt from CHANGELOG changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog
[test-exempt]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#tests

### CODE REVIEW

**Review by johnpryan — APPROVED** [review]

File: packages/go_router/lib/src/information_provider.dart [hunk 1]
```diff
@@ -79,18 +80,35 @@ class GoRouteInformationProvider extends RouteInformationProvider
...
-        _valueInEngine = _kEmptyRouteInformation {
+        _valueInEngine = _kEmptyRouteInformation,
+        _routerNeglect = routerNeglect {
     _refreshListenable?.addListener(notifyListeners);
   }
 
+  /// Notifies the platform for a route information change.
+  @visibleForTesting
+  @internal
+  Future<void> systemNavigatorRouteInformationUpdated(
```
Comment (chunhtai) [inline packages/go_router/lib/src/information_provider.dart:98]: This is not necessary, you can listen to platform message during test using
See example here https://github.com/flutter/flutter/blob/d39550fb528c863431f1e3b754d4279defd82081/packages/flutter/test/services/system_navigator_test.dart#L34

Comment (tolo) [reply packages/go_router/lib/src/information_provider.dart:98]: Ah, great, will update the code.

File: packages/go_router/lib/src/builder.dart [hunk 2]
```diff
@@ -98,7 +98,8 @@ class RouteBuilder {
   Widget build(
     BuildContext context,
     RouteMatchList matchList,
-    bool routerNeglect,
+    bool routerNeglect, // TODO(tolo): This parameter is not used and should be
```
Comment (chunhtai) [inline packages/go_router/lib/src/builder.dart:101]: Can you add a doc somewhere, probably in configuration.md?

**Review by chunhtai — APPROVED** [review]
just a request to add some doc around this parameter, otherwise LGTM

File: packages/go_router/lib/src/builder.dart (hunk 2 above)
Comment (chunhtai) [reply packages/go_router/lib/src/builder.dart:101]: I meant to leave the comment in GoRouter, but yeah just document on what this routerNeglect now does (basically turn it in to a single page app without history)

**Comment by chunhtai** [comment]
wonderful thank you

**Review by chunhtai — APPROVED** [review]
still LGTM

**Pull request merged**